asdf_astropy/converters/fits/tests/test_fits.py

```diff
@@ -0,0 +1,105 @@
+import asdf
+from asdf.tests.helpers import yaml_to_asdf
+from astropy.io import fits
+import numpy as np
+from numpy.testing import assert_array_equal
+import pytest
+
+from asdf_astropy.testing.helpers import assert_hdu_list_equal
+
+
+HDUL = fits.HDUList()
+HEADER = fits.Header([
+    ("FOO", "BAR", "BAZ"),
+    ("SOMENUM", "11.0"),
+    ("EMPTY",)
+])
+HDUL.append(fits.PrimaryHDU(header=HEADER))
+HDUL.append(fits.ImageHDU(data=np.arange(100)))
+
+HDUL_WITH_TABLE = fits.HDUList()
+HDUL_WITH_TABLE.append(
+    fits.BinTableHDU.from_columns(
+        np.array([(0, 1), (2, 3)], dtype=[("A", int), ("B", int)])
+    )
+)
+
+TEST_HDULS = [
+    HDUL,
+    HDUL_WITH_TABLE,
+]
```

Comment:
This defines a lot of global variables.  Could this instead be something like:

```
def _generate_test_hdulists():
    hdul = fits.HDUList()
    ...


@pytest.mark.parametrize("hdul", _generate_test_hdulists())
def test_serialization(hdul, tmp_path):
    ...
```

Reply:
Hmm... your suggestion also has the benefit of creating new objects for each test which better isolates them.  I used these global variables in all the test files, I'll make the suggested change everywhere.

Reply:
... but how about in a separate PR, so reviewers of this one don't have to wade through it all

Reply:
No worries.  👍

Reply:
Here's the separate PR: https://github.com/astropy/asdf-astropy/pull/25